src/components/serversContext/serversContext.tsx

```diff
@@ -0,0 +1,251 @@
+import { Repo } from "@automerge/automerge-repo";
+import { BrowserWebSocketClientAdapter } from "@automerge/automerge-repo-network-websocket";
+import { IndexedDBStorageAdapter } from "@automerge/automerge-repo-storage-indexeddb";
+import { isEqual } from "lodash";
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { STORAGE_KEYS } from "../../lib/constants/localStorageKeys";
+import { TDocType } from "../../types/schema";
+import { openDeleteConfirmModal } from "../modals/modals";
+import {
+  REPOSITORY_TYPE,
+  SERVER_STATUS,
+  ServersList,
+  StorageServersConfig,
+  TServersContextValue,
+  TServersProviderProps,
+  YtServer,
+} from "./types";
+
+const urlPrefix = "automerge:";
+
+let done = false;
```

Comment:
Could we use something like Mantine's [useIsFirstRender](https://mantine.dev/hooks/use-is-first-render/) instead of this `done` variable?